qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -66,67 +67,67 @@ def copy(self):
         ret._analyses = [analysis.copy() for analysis in ret._analyses]
         return ret
 
+    def run(
+        self,
+        experiment_data: ExperimentData,
+        replace_results: bool = False,
+        **options,
+    ) -> ExperimentData:
+        # Make a new copy of experiment data if not updating results
+        if not replace_results and _requires_copy(experiment_data):
+            experiment_data = experiment_data.copy()
+
+        # Initialize child components if they are not initalized.
+        self._initialize_child_data(experiment_data)
+
+        # Run analysis with replace_results = True since we have already
+        # created the copy if it was required
+        return super().run(experiment_data, replace_results=True, **options)
+
     def _run_analysis(self, experiment_data: ExperimentData):
         # Return list of experiment data containers for each component experiment
         # containing the marginalied data from the composite experiment
-        component_exp_data = self._component_experiment_data(experiment_data)
+        component_expdata = self._component_experiment_data(experiment_data)
```

Comment:
Should `_component_experiment_data` be renamed now because `component_experiment_data` attribute has been already renamed in the `ExperimentData`?